solana/solana-ibc/programs/solana-ibc/src/lib.rs

```diff
@@ -505,6 +588,19 @@ pub struct SendPacket<'info> {
     system_program: Program<'info, System>,
 }
 
+#[derive(Accounts)]
+pub struct FormMessageChunks<'info> {
+    #[account(mut)]
+    sender: Signer<'info>,
+
+    #[account(init_if_needed, payer = sender, seeds = [MSG_CHUNKS], bump, space = 10240)]
+    pub msg_chunks: Account<'info, storage::MsgChunks>,
```

Comment:
Two things.

First of all, do we need this to be a PDA? I was actually wondering if it wouldn’t be better if this was just an account owned by the sender and then they can manage it in any way they want.  All we need here is a writeable account.

Second of all, if this is a PDA it *must* include sender in the seeds.  Without that, users will overwrite each others calls.

Reply:
yes i was thinking about it too. I was also looking if we could create an account with data outside of smart contract. Couldnt find anything as now. But we dont need PDA, we could use the sender's account too. Once the `deliver_chunks` is called, we can also close the account so that we can return the rent.